File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/main/chunk/storage/ChunkPos.java
package pize.tests.voxelgame.main.chunk.storage;

import java.util.Objects;

import static pize.tests.voxelgame.main.chunk.ChunkUtils.SIZE;

public class ChunkPos{

    public final int x, z;

    public ChunkPos(int x, int z){
        this.x = x;
        this.z = z;
    }
    

    public ChunkPos getNeighbor(int x, int z){
        return new ChunkPos(this.x + x, this.z + z);
    }
    
    public int globalX(){
        return x * SIZE;
    }
    
    public int globalZ(){
        return z * SIZE;
    }


    @Override
    public boolean equals(Object object){
        if(object == this)
            return true;
        if(object == null || object.getClass() != getClass())
            return false;
        ChunkPos chunkPos = (ChunkPos) object;
        return x == chunkPos.x && z == chunkPos.z;
    }

    @Override
    public int hashCode(){
        return Objects.hash(x, z);
    }
    
    @Override
    public String toString(){
        return x + ", " + z;
    }

}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/main/chunk/storage/ChunkPos.kt
package pize.tests.voxelgame.main.chunk.storage

import pize.tests.voxelgame.main.chunk.ChunkUtils
import java.util.*

class ChunkPos(val x: Int, val z: Int) {
    fun getNeighbor(x: Int, z: Int): ChunkPos {
        return ChunkPos(this.x + x, this.z + z)
    }

    fun globalX(): Int {
        return x * ChunkUtils.SIZE
    }

    fun globalZ(): Int {
        return z * ChunkUtils.SIZE
    }

    override fun equals(`object`: Any?): Boolean {
        if (`object` === this) return true
        if (`object` == null || `object`.javaClass != javaClass) return false
        val chunkPos = `object` as ChunkPos
        return x == chunkPos.x && z == chunkPos.z
    }

    override fun hashCode(): Int {
        return Objects.hash(x, z)
    }

    override fun toString(): String {
        return "$x, $z"
    }
}
